File: fix-icon-imports.js
const fs = require('fs');
const path = require('path');

// Function to fix icon imports and usage
function fixIconImports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;

    // Check if file has data arrays with string icon names
    if (content.includes("icon: 'Users'") || content.includes("icon: 'Clock'") || content.includes("icon: 'Zap'")) {
      // Get all unique icon names from the data arrays
      const iconMatches = content.match(/icon: '([^']+)'/g);
      if (iconMatches) {
        const iconNames = [...new Set(iconMatches.map(match => match.match(/icon: '([^']+)'/)[1]))];
        
        // Add missing icon imports
        const currentLucideImport = content.match(/import { [^}]+ } from 'lucide-react';/);
        if (currentLucideImport) {
          const currentImports = currentLucideImport[0].match(/{ ([^}]+) }/)[1];
          const newImports = currentImports + ', ' + iconNames.join(', ');
          content = content.replace(currentLucideImport[0], `import { ${newImports} } from 'lucide-react';`);
        } else {
          content = content.replace(
            /import { ArrowRight[^}]* } from 'lucide-react';/,
            `import { ArrowRight, ${iconNames.join(', ')} } from 'lucide-react';`
          );
        }

        // Replace string icon names with actual icon components
        iconNames.forEach(iconName => {
          const iconComponent = iconName.charAt(0).toUpperCase() + iconName.slice(1);
          content = content.replace(new RegExp(`icon: '${iconName}'`, 'g'), `icon: ${iconComponent}`);
        });

        modified = true;
      }
    }

    // Add Footer import if missing
    if (content.includes('<Footer') && !content.includes("import Footer from")) {
      content = content.replace(
        /import { [^}]+ } from 'lucide-react';/,
        `import { ${content.match(/import { ([^}]+) } from 'lucide-react';/)[1]} } from 'lucide-react';\nimport Footer from '@/components/Footer';`
      );
      modified = true;
    }

    // Fix missing closing tags
    if (content.includes('<Footer') && !content.includes('</>')) {
      content = content.replace(/(<Footer[^>]*\/>)\s*$/, '$1\n    </>');
      modified = true;
    }

    if (modified) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${filePath}`);
      return true;
    }
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
  }
  return false;
}

// Function to recursively find all .tsx files
function findTsxFiles(dir) {
  const files = [];
  const items = fs.readdirSync(dir);
  
  for (const item of items) {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
      files.push(...findTsxFiles(fullPath));
    } else if (item.endsWith('.tsx')) {
      files.push(fullPath);
    }
  }
  
  return files;
}

// Main execution
const appDir = '/workspace/app';
const tsxFiles = findTsxFiles(appDir);

console.log(`Found ${tsxFiles.length} .tsx files to check`);

let fixedCount = 0;
for (const file of tsxFiles) {
  if (fixIconImports(file)) {
    fixedCount++;
  }
}

console.log(`Fixed ${fixedCount} files`);

File: fix_icon_imports.py
#!/usr/bin/env python3
"""
Script to fix missing Lucide React icon imports across all pages.
"""

import os
import re
from pathlib import Path
from collections import defaultdict

# Common Lucide React icons used in the project
ICON_MAPPING = {
    'CheckCircle': 'CheckCircle',
    'ArrowRight': 'ArrowRight',
    'Phone': 'Phone',
    'Mail': 'Mail',
    'MapPin': 'MapPin',
    'Zap': 'Zap',
    'Shield': 'Shield',
    'Brain': 'Brain',
    'Globe': 'Globe',
    'Star': 'Star',
    'Camera': 'Camera',
    'Eye': 'Eye',
    'BarChart': 'BarChart',
    'Target': 'Target',
    'TrendingUp': 'TrendingUp',
    'Users': 'Users',
    'Settings': 'Settings',
    'Database': 'Database',
    'Code': 'Code',
    'Cloud': 'Cloud',
    'Heart': 'Heart',
    'Palette': 'Palette',
    'Music': 'Music',
    'Video': 'Video',
    'DollarSign': 'DollarSign',
    'Calendar': 'Calendar',
    'Clock': 'Clock',
    'Bot': 'Bot',
    'Cpu': 'Cpu',
    'PhoneIcon': 'Phone',
    'MailIcon': 'Mail',
    'Check': 'Check',
    'Helmet': 'Helmet'
}

def find_used_icons(content):
    """Find all Lucide React icons used in the content."""
    used_icons = set()
    
    # Look for JSX elements that are likely Lucide icons
    icon_pattern = r'<(\w+)(?:\s|>)'
    matches = re.findall(icon_pattern, content)
    
    for match in matches:
        if match in ICON_MAPPING:
            used_icons.add(ICON_MAPPING[match])
    
    return used_icons

def fix_icon_imports(file_path):
    """Fix missing Lucide React icon imports in a file."""
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Skip if not a React component file
    if not ('import React' in content or 'export default' in content):
        return False
    
    # Find used icons
    used_icons = find_used_icons(content)
    
    if not used_icons:
        return False
    
    # Check if lucide-react is already imported
    lucide_import_match = re.search(r"import\s*{\s*([^}]+)\s*}\s*from\s*['\"]lucide-react['\"]", content)
    
    if lucide_import_match:
        # Update existing import
        existing_icons = [icon.strip() for icon in lucide_import_match.group(1).split(',')]
        all_icons = list(set(existing_icons + list(used_icons)))
        all_icons.sort()
        
        new_import = f"import {{ {', '.join(all_icons)} }} from 'lucide-react'"
        content = content.replace(lucide_import_match.group(0), new_import)
    else:
        # Add new import after React import
        react_import_match = re.search(r"import\s+React[^;]+;", content)
        if react_import_match:
            new_import = f"\nimport {{ {', '.join(sorted(used_icons))} }} from 'lucide-react'"
            content = content.replace(react_import_match.group(0), react_import_match.group(0) + new_import)
        else:
            # Add at the top
            new_import = f"import {{ {', '.join(sorted(used_icons))} }} from 'lucide-react'\n"
            content = new_import + content
    
    # Write the updated content
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    return True

def fix_unescaped_entities(file_path):
    """Fix unescaped entities in JSX."""
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Fix unescaped quotes and apostrophes
    content = re.sub(r"'([^']*)'", r"&apos;\1&apos;", content)
    content = re.sub(r'"([^"]*)"', r"&quot;\1&quot;", content)
    
    # Write the updated content
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    return True

def main():
    """Main function to fix all icon imports and unescaped entities."""
    
    # Find all React component files
    component_files = []
    
    # Search in app directory
    app_dir = Path('/workspace/app')
    if app_dir.exists():
        component_files.extend(app_dir.rglob('*.tsx'))
    
    # Search in src directory
    src_dir = Path('/workspace/src')
    if src_dir.exists():
        component_files.extend(src_dir.rglob('*.tsx'))
    
    icon_fixes = 0
    entity_fixes = 0
    
    for file_path in component_files:
        try:
            if fix_icon_imports(str(file_path)):
                icon_fixes += 1
                print(f"Fixed icon imports in: {file_path}")
            
            if fix_unescaped_entities(str(file_path)):
                entity_fixes += 1
                print(f"Fixed unescaped entities in: {file_path}")
                
        except Exception as e:
            print(f"Error processing {file_path}: {e}")
    
    print(f"\nFixed icon imports in {icon_fixes} files")
    print(f"Fixed unescaped entities in {entity_fixes} files")

if __name__ == '__main__':
    main()